src/CMakeLists.txt

```diff
@@ -87,6 +87,43 @@ target_compile_definitions(dlaf.prop
     $<$<BOOL:${DLAF_WITH_CUDA_MPI_RDMA}>:DLAF_WITH_CUDA_MPI_RDMA>
 )
 
+# Precompiled headers
+if(DLAF_WITH_PRECOMPILED_HEADERS)
+  add_library(dlaf.pch_lib OBJECT dummy.cpp)
+  target_link_libraries(dlaf.pch_lib PRIVATE dlaf.prop)
+  target_add_warnings(dlaf.pch_lib)
+
+  add_executable(dlaf.pch_exe dummy.cpp)
+  target_link_libraries(dlaf.pch_exe PRIVATE dlaf.prop)
+  target_add_warnings(dlaf.pch_exe)
```

Comment:
There are two precompiled headers because executables and libraries may be compiled with different flags making e.g. a library precompiled header incompatible with executables. In my case executables were compiled with `-fpie` and libraries not.

Reply:
Have you considered using an INTERFACE library instead of a classic library with a `dummy.cpp`?

Reporting also from the [doc](https://cmake.org/cmake/help/v3.16/command/target_precompile_headers.html?highlight=target_precompile_header), it looks like our use case
> [...] A notable exception to this is where an [interface library](https://cmake.org/cmake/help/v3.16/manual/cmake-buildsystem.7.html#interface-libraries) is created to define a commonly used set of precompile headers in one place and then other targets link to that interface library privately. In this case, the interface library exists specifically to propagate the precompile headers to its consumers and the consumer is effectively still in control, since it decides whether to link to the interface library or not.

Moreover, I'm missing the point of having `add_executable(dlaf.pch_exe dummy.cpp)` and then use it as a library. Why not just `add_library` here too?

Reply:
Thanks for pointing that out! I had a vague memory of it not working as expected with interface libraries (probably because of wrong flags again) but I wasn't aware of this part of the docs and I might have done it wrong so I'll have another look. That would definitely be nicer.

The separate target for executables is there also for the correct flags but if using an interface library gets rid of that need then it's also not necessary to have a separate target.

Note that `dlaf.pch_exe` is _not_ used like a library which might be causing some of the confusion. Or I'm interpreting "use it as a library" differently than you mean.

Reply:
I will add some more comments into the code about this, but in short we can't (really) use interface libraries here.

This CMake issue is relevant: https://gitlab.kitware.com/cmake/cmake/-/issues/20288. Particularly this comment: https://gitlab.kitware.com/cmake/cmake/-/issues/20288#note_689092.

In short: one can add precompiled headers to interface libraries but they will also only be interface precompiled headers and no precompiled header will be generated for the interface target. This means one has to link to the interface library, not "reuse" the precompiled header from the library and all targets linking to this will have a separate precompiled header which is unnecessary.

Reply:
Added a comment about this. Please have a look.